src/eetcd.erl

```diff
@@ -11,19 +11,21 @@
 %% `open(test,["127.0.0.1:2379","127.0.0.1:2479","127.0.0.1:2579"]).'
 -spec open(name(), [string()]) -> {ok, pid()} | {error, any()}.
 open(Name, Hosts) ->
-    open(Name, Hosts, [], tcp, []).
+    open(Name, Hosts, [], tcp, [], []).
 
 %% @doc Connects to a etcd server.
 -spec open(name(),
     [string()],
     tcp | tls | ssl,
-    [gen_tcp:connect_option()] | [ssl:connect_option()]) ->
+    [gen_tcp:connect_option()],
+    [ssl:tls_client_option()]) ->
     {ok, pid()} | {error, any()}.
-open(Name, Hosts, Transport, TransportOpts) ->
-    open(Name, Hosts, [], Transport, TransportOpts).
+open(Name, Hosts, Transport, TcpOpts, TlsOpts) when is_atom(Transport) ->
+    open(Name, Hosts, [], Transport, TcpOpts, TlsOpts).
```

Comment:
It is a little strange to put `TcpOps` and  `TlsOpts` together into one function.

- `TcpOpts` is associated with the transport `tcp`
- `TlsOpts` is associated with the transport `ssl` or `tls`

Maybe this way is a little better:

I think it is better to combine `Transport` and `TransportOpts` into one tuple and place it into the `Options`.

```erlang
-spec open(Name, Hosts, Options)  -> {ok, pid()} | {error, any()} when
    Name :: name(),
    Hosts :: [string()],
    Options :: [ {mode, connect_all | random} |
                        {name, string()} |
                        {password, string()}  |
                        {retry, non_neg_integer()} |
                        {retry_timeout, pos_integer()} |
                        {connect_timeout, timeout()} 
                        {transport, tcp | ssl | tls} |
                        {transport_opts, [gen_tcp:connect_option()] |  [ssl:tls_client_option()]}
                      ].
```

Reply:
Regardless, this pull request will introduce a breaking change. I believe we can offer a simpler API in this case.

Reply:
That makes sense. I have tried to make this change, and it looks good.